Lib/test/test_imaplib.py

```diff
@@ -537,6 +603,17 @@ def test_unselect(self):
         self.assertEqual(data[0], b'Returned to authenticated state. (Success)')
         self.assertEqual(client.state, 'AUTH')
 
+    # property tests
+
+    def test_file_property(self):
+        client, _ = self._setup(SimpleIMAPHandler)
+        # 'file' attribute access should trigger a warning
+        with warnings.catch_warnings(record=True) as warned:
+            warnings.simplefilter('always')
+            client.file
+            self.assertEqual(len(warned), 1)
+            self.assertTrue(issubclass(warned[-1].category, RuntimeWarning))
```

Comment:
This construction works but you might want to have a look at [docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns) for a possible simpler alternative.